pupil_src/shared_modules/plugin.py

```diff
@@ -385,6 +367,42 @@ def add(self, new_plugin, args={}):
         if self.g_pool.app in ("capture", "player"):
             plugin_instance.init_ui()
 
+    def _avoid_duplication(self, new_plugin_cls):
```

Comment:
I'd call this `_remove_duplicates(...)`.
`avoid` does not really tell what's happening.